packages/better-auth/src/plugins/app-invite/index.ts

```diff
@@ -0,0 +1,860 @@
+	/**
+	 * Define wheter a user is allowed to cancel invitations.
+	 *
+	 * By default users can only cancel invitations issued by themself.
+	 */
+	allowUserToCancelInvitation?: (data: {
+		user: User;
+		invitation: AppInvitation;
+	}) => Promise<boolean> | boolean;
+	/**
+	 * The expiration time for the invitation link.
+	 *
+	 * @default 48 hours
+	 */
+	invitationExpiresIn?: number;
+	/**
+	 * Send an email with the
+	 * invitation link to the user.
+	 */
+	sendInvitationEmail?: (
+		data: {
+			/**
+			 * the invitation id
+			 */
+			id: string;
+			/**
+			 * the email of the user
+			 */
+			email: string;
+			/**
+			 * the member who is inviting the user
+			 */
+			inviter: User;
+		},
+		request?: Request,
+	) => Promise<void>;
+	/**
+	 * The schema for the app-invite plugin.
+	 */
+	schema?: {
+		appInvitation?: {
+			modelName?: string;
+			fields?: {
+				[key in keyof Omit<AppInvitation, "id">]?: string;
+			};
+		};
+	};
+	/**
+	 * Automatically sign in the user after sign up
+	 */
+	autoSignIn?: boolean;
+	/**
+	 * Rate limit configuration.
+	 *
+	 * @default {
+	 *  window: 60,
+	 *  max: 5,
+	 * }
+	 */
+	rateLimit?: {
+		window: number;
+		max: number;
+	};
+	$Infer?: {
+		/**
+		 * Infer additional fields for the user
+		 */
+		AdditionalFields?: Record<string, any>;
+	};
+}
+
+/**
+ * App Invite plugin for Better Auth. This plugin allows you to invite other users the the app.
+ *
+ * @example
+ * ```ts
+ * const auth createAuth({
+ *  plugins: [
+ *      appInvite({
+ *          sendInvitationEmail: async (data) => {
+ *              // ...
+ *          }
+ *      })
+ *  ]
+ * })
+ * ```
+ */
+export const appInvite = <O extends AppInviteOptions>(opts?: O) => {
+	const options = {
+		allowUserToCreateInvitation: true,
+		allowUserToCancelInvitation: ({ user, invitation }) => {
+			return invitation.inviterId === user.id;
+		},
+		...opts,
+	} satisfies AppInviteOptions;
+
+	return {
+		id: "app-invite",
+		endpoints: {
+			createAppInvitation: createAuthEndpoint(
+				"/invite-user",
+				{
+					method: "POST",
+					use: [sessionMiddleware],
+					body: z.object({
+						email: z
+							.string({
+								description: "The email address of the user to invite",
+							})
+							.optional(),
+						resend: z
+							.boolean({
+								description:
+									"Resend the invitation email, if the user is already invited",
+							})
+							.optional(),
+						domainWhitelist: z
+							.string({
+								description:
+									"A comma separated list of domains that are allowed to accept the invitation. (Only available for public invitations)",
+							})
+							.optional(),
+					}),
+					metadata: {
+						openapi: {
+							description: "Invite a user to the app",
+							responses: {
+								"200": {
+									description: "Success",
+									content: {
+										"application/json": {
+											schema: {
+												type: "object",
```

Comment:
you could use the $ref like `$ref: "#/components/schemas/AppInvitation",`

same comment for other methods